cgi-bin/DW/Controller/Customize.pm

```diff
@@ -0,0 +1,654 @@
+    my $theme_html = DW::Template->template_string( 'customize/themechooser.tt',
+        { 'theme_data' => $theme_data, qargs => $queryargs } );
+    my $current_data = get_current_data(0);
+
+    $r->print( to_json( { theme_html => $theme_html, current_data => $current_data } ) );
+    return $r->OK;
+}
+
+sub layoutchooser_handler {
+
+    # gets the request and args
+    my $r    = DW::Request->get;
+    my $args = $r->post_args;
+
+    # set the new titles
+
+    set_layout($args);
+
+    $r->print( to_json( get_layout_data($args) ) );
+    return $r->OK;
+}
+
+sub journaltitles_handler {
+
+    # gets the request and args
+    my $r    = DW::Request->get;
+    my $args = $r->post_args;
+
+    # set the new titles
+
+    set_journaltitles($args);
+
+    return $r->OK;
+}
+
+sub options_handler {
+    my ( $ok, $rv ) = controller( authas => 1, form_auth => 1 );
+    return $rv unless $ok;
+
+    my $r      = $rv->{r};
+    my $POST   = $r->post_args;
+    my $u      = $rv->{u};
+    my $remote = $rv->{remote};
+    my $GET    = $r->get_args;
+
+    # if using s1, switch them to s2
+    unless ( $u->prop('stylesys') == 2 ) {
+        $u->set_prop( stylesys => 2 );
+    }
+
+    my $group = $GET->{group} ? $GET->{group} : "presentation";
+
+    # make sure there's a style set and load it
+    my $style = LJ::Customize->verify_and_load_style($u);
+
+    # lazy migration of style name
+    LJ::Customize->migrate_current_style($u);
+
+    my $vars;
+    $vars->{u}            = $u;
+    $vars->{remote}       = $remote;
+    $vars->{is_identity}  = 1 if $u->is_identity;
+    $vars->{is_community} = 1 if $u->is_community;
+    $vars->{style}        = LJ::Customize->verify_and_load_style($u);
+    $vars->{authas_html}  = $rv->{authas_html};
+
+    # pass our computed values to the template
+    $vars->{help_icon} = \&LJ::help_icon;
+
+    my $customize_theme = LJ::Widget::CustomizeTheme->new;
+    my $headextra       = $customize_theme->wrapped_js( page_js_obj => "Customize" );
+    my $ret             = "<div class='customize-wrapper one-percent'>";
+    $ret .= $customize_theme->render(
+        group     => $group,
+        headextra => \$headextra,
+        post      => $POST,
+    );
+    $ret .= "</div><!-- end .customize-wrapper -->";
+
+    #handle post actions
+
+    if ( LJ::did_post() ) {
+        my @errors = LJ::Widget->handle_post( $POST,
+            qw(CustomizeTheme CustomTextModule MoodThemeChooser NavStripChooser S2PropGroup LinksList)
+        );
+        $ret .= LJ::bad_input(@errors) if @errors;
+    }
+
+    $vars->{content}     = $ret;
+    $vars->{layout_data} = get_layout_data();
+
+    $vars->{current_data} = get_current_data(1);
+
+    # Now we tell it what template to render and pass in our variables
+    return DW::Template->render_template( 'customize/index.tt', $vars );
+
+}
+
+## Functions to handle edits for both RPC and main endpoints
+
+sub set_theme {
+    my %opts = @_;
+    my $u    = LJ::get_effective_remote();
```

Comment:
I know this is pretty common in BML but since we moved to Controllers, I'd prefer we pass in the `$u` we're operating on instead of inferring it from the environment. Do we do this in other controllers?

Reply:
Just checked and yeah, there is only one single use of `get_effective_remote` in the new controllers (in Create). We do seem to use `LJ::get_remote();` for RPC endpoints though - I will see if the regular Controller formulation works for that or not.